lib/src/iterable_extensions.dart

```diff
@@ -56,6 +56,27 @@ extension IterableExtension<T> on Iterable<T> {
     return chosen;
   }
 
+  /// Separates the elements of the iterable with the specified separator.
+  ///
+  /// The [separated] method takes a separator as input and returns a new
+  /// iterable where the elements are separated by the specified separator.
+  ///
+  /// Example:
+  /// ```dart
+  /// final list = [1, 2, 3, 4];
+  /// final separatedList = list.separated(0);
+  /// print(separatedList); // Output: [1, 0, 2, 0, 3, 0, 4]
+  /// ```
+  Iterable<T> separated(T separator) sync* {
+    final iterator = this.iterator;
+    if (!iterator.moveNext()) return;
+    yield iterator.current;
+    while (iterator.moveNext()) {
+      yield separator;
+      yield iterator.current;
+    }
+  }
```

Comment:
I don't think this version is flexible enough.
The use-cases I've heard mentioned won't all be satisfies by inserting the *same* value every time, and the typing might not work out.

I'd *at least* want a version where the separator is generated each time.
Then it could be generated based on position, or on adjacent values.

Something like these would cover those:
```dart
Iterable<T> separatedByValue(T separator) { ... }
Iterable<T> separatedBy(T Function() separator) { ... }
```
and possibly also:
```dart
// Position is index of next element, from 1 to `lenght` - 1.
Iterable<T> separatedByIndex(T Function(int position) separator) { ... }
Iterable<T> separatedByElements(T Function(T before, T after) separator) { ... }
```
The `separatedBy` and `separatedByIndex` might be combined into one, it's easy to ignore an argument, and cheap to increment an integer.

I'd expect the separator to not always have the same type as the elements being separated. (A `List<TextWidgets>` cannot be separated by `SeparatorWidget`s, one would have to first copy the text widgets into a `List<Widget>` and then separate them)

That makes a top-level function more useful (or stastic function or constructor, if we could add such using extensions):
```dart
  static Iterable<S> separatedBy<S>(Iterable<S> elements, S Function(int position)  separator) { ... }
  static Iterable<S> separatedByValue<S>(Iterable<S> elements, S  separator) { ... }
  static Iterable<S> separatedByElements<S, T extends S>(Iterable<T> elements, S Function(T before, T after)  separator) { ... }
```
Maybe we can make the separator optional, like: 
```dart
  static Iterable<S> separatedOptionalBy<S>(Iterable<S> elements, S? Function(int position)  separator) { ... }
  static Iterable<S> separatedOptionalByElements<S, T extends S>(Iterable<T> elements, S? Function(T before, T after)  separator) { ... }
```
where returning `null` means no separator.
(Consider: `separatedOptionalByElements(pathSegments, (before, after) => before.endsWith('/') || after.startsWith('/') ? null : '/')` where we only insert a separator where it's needed. Maybe this is too specialized, and a `join` with  conditional separator would be more relevant.)